povm_toolbox/post_processor/povm_post_processor.py

```diff
@@ -15,28 +15,47 @@
 import numpy as np
 from qiskit.quantum_info import SparsePauliOp
 
+from povm_toolbox.quantum_info.base_dual import BaseDUAL
+from povm_toolbox.quantum_info.multi_qubit_dual import MultiQubitDUAL
+from povm_toolbox.quantum_info.multi_qubit_povm import MultiQubitPOVM
+from povm_toolbox.quantum_info.product_dual import ProductDUAL
+from povm_toolbox.quantum_info.product_povm import ProductPOVM
 from povm_toolbox.sampler import POVMPubResult
 
 
-class POVMPostprocessor:
+class POVMPostProcessor:
     """A common POVM result post-processor."""
 
     def __init__(
         self,
         povm_sample: POVMPubResult,
-        alphas: np.ndarray | None = None,
+        DUAL_CLASS: type[BaseDUAL] | None = None,
     ) -> None:
         """Initialize the POVM post-processor.
 
         Args:
             povm_sample: a result from a POVM sampler run.
-            alphas: parameters of the frame superoperator of the POVM.
+            alphas: parameters of the frame super-operator of the POVM.
         """
         self.povm = povm_sample.metadata.povm_implementation.definition()
         self.counts: np.ndarray = povm_sample.get_counts()  # type: ignore
         # TODO: find a way to avoid the type ignore
-        if alphas is not None:
-            self.povm.alphas = alphas
+
+        if DUAL_CLASS is None:
+            if isinstance(self.povm, MultiQubitPOVM):
+                DUAL_CLASS = MultiQubitDUAL
+            elif isinstance(self.povm, ProductPOVM):
+                DUAL_CLASS = ProductDUAL
+            else:
+                raise TypeError
+        elif not issubclass(DUAL_CLASS, BaseDUAL):
+            raise TypeError
```

Comment:
I don't think this is particularly good design to centralize the `BasePOVM -> BaseDUAL` mapping inside of the `POVMPostProcessor`...

Can we instead add a class attribute to the `BasePOVM` which points to the corresponding `BaseDUAL` class?
Something along the lines of:
```python

class BasePOVM:

    dual_class = BaseDUAL

class MultiQubitPOVM(BasePOVM):

    dual_class = MultiQubitDUAL


# and so on
```